eeprom.c

```diff
@@ -451,6 +451,7 @@ eeprom_write (eeprom_context_t ctx, module_eeprom_t *data)
 		memcpy(rawdata->system_serialnumber_v2, data->system_serialnumber,
 		       sizeof(rawdata->system_serialnumber_v2));
 	}
+	rawdata->length = sizeof(*rawdata) - 1;
```

Comment:
Could you use `htole16()` here, as I've done elsewhere for 16-bit fields?

Also, could you update the 'no longer used' comment on the `length` field at the top of the file, it's obviously wrong. :)